src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/ValidateArgumentsOfPublicMethodsTests.cs

```diff
@@ -6715,5 +6715,54 @@ public void M(object instance)
 }",
             }.RunAsync();
         }
+
+        [Fact, WorkItem(3634, "https://github.com/dotnet/roslyn-analyzers/issues/3634")]
+        public async Task NullConditionalAssignmentOperator_NullableEnableContext_NoDiagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+#nullable enable
+using System.Collections.Generic;
+
+public class C
+{
+    public void ParameterTest(Dictionary<string, string>? dict = null)
+    {
+        dict ??= new Dictionary<string, string>();
+        SetParameter(dict);
```

Comment:
Consider adding another test that uses `SetParameter(dict ?? new Dictionary<string, string>());`

Reply:
I have just checked and we already have some tests covering such use case (the untested bit was the null coalesce assignment or whatever the official name of the operator is xD).